assets/src/apps/page-editor/PageEditor.tsx

```diff
@@ -551,32 +560,34 @@ export class PageEditor extends React.Component<PageEditorProps, PageEditorState
             </Objectives>
 
             <div className="d-flex flex-row">
-              <ContentOutline
-                editMode={this.state.editMode}
-                content={this.state.content}
-                activityContexts={this.state.activityContexts}
-                editorMap={props.editorMap}
-                projectSlug={projectSlug}
-                resourceSlug={resourceSlug}
-                onEditContent={onEdit}
-              />
-              <Editors
-                {...props}
-                editMode={this.state.editMode}
-                objectives={this.state.allObjectives}
-                allTags={this.state.allTags}
-                childrenObjectives={this.state.childrenObjectives}
-                onRegisterNewObjective={onRegisterNewObjective}
-                onRegisterNewTag={onRegisterNewTag}
-                activityContexts={this.state.activityContexts}
-                onRemove={(key: string) => this.onRemove(key)}
-                onEdit={onEdit}
-                onEditActivity={this.onEditActivity}
-                onPostUndoable={this.onPostUndoable}
-                content={this.state.content}
-                onAddItem={onAddItem}
-                resourceContext={props}
-              />
+              <AlternativesContextProvider projectSlug={projectSlug}>
```

Comment:
Alternative options are loaded at the top-level so that there is only a single network request and when the request resolves (or fails) all components are up to date at the same time.